### ISSUE TITLE
Can't delete primary email

### ISSUE THREAD

**Issue by pjreddie** [pennersr/django-allauth#290]
Even if an email is not required, you can't delete a primary email account and you can't make the email not primary unless you have one. So if you happen to click make primary once, you are forever stuck with some email (even if it isn't real) instead of having the option to delete all email addresses. It just seems like a strange design choice.

**Comment by numpde** [comment]
I second the "strange design choice" opinion.  The allauth framework is imposing too many constraints here.  I want my visitors to be able to remove all e-mail if they want.  With notifications and social login it is not necessary to have one and feels authoritarian.  Thank you.

**Comment by 9mido** [comment]
Views.py EmailView class _action_remove function:

```
            unverified_primary_email = EmailAddress.objects.filter(user=request.user,email=email, primary=True, verified=False).values_list('email', flat=True)
            if email_address.primary and email not in unverified_primary_email:
                get_adapter(request).add_message(
                    request,
                    messages.ERROR,
                    'account/messages/'
                    'cannot_delete_primary_email.txt',
                    {"email": email})
            elif email in unverified_primary_email:
                email_address.delete()
            else:
                email_address.delete()
```

**Comment by derek-adair** [comment]
Close as per discussion in https://github.com/pennersr/django-allauth/pull/2774 (i think??).  It seems like this is a corner case that can be solved w/ a custom adapter.

**Comment by pennersr** [comment]
No, this is a valid issue. If `ACCOUNT_EMAIL_REQUIRED=False` and `ACCOUNT_AUTHENTICATION_METHOD="username"` then indeed you should be able to delete all email addresses from your account.

**Comment by derek-adair** [comment]
Right but is that not solved by a custom adapter? Or did i misread your comment to mean it's worth modifying the core adapter to support this?

**Comment by pennersr** [comment]
It is an issue that can be worked around with by customizing things yourself, but this should indeed just work out of the box.

**Comment by pennersr** [comment]
Closed via a0ff1aa2

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/290
